components/turret.py

```diff
@@ -107,110 +81,70 @@ def scan(self, heading):
         # Otherwise scan about the heading we've been given.
         # The target must be downfield from us, so scan up to
         # 90 degrees either side of the given heading
-        pass
+
+        # First reset scan size
+        self.current_scan_increment = self.scan_increment_degrees
+        if self.index_found:
+            # set the first pass
+            self._slew_to_counts(
+                (azimuth + math.radians(self.scan_increment_degrees))
+                * self.COUNTS_PER_TURRET_RADIAN
+            )
+        else:
+            current_count = self.motor.getSelectedSensorPosition()
+            self._slew_to_counts(
+                current_count
+                + (
+                    math.radians(self.scan_increment_degrees)
+                    * self.COUNTS_PER_TURRET_RADIAN
+                )
+            )
 
     def is_ready(self) -> bool:
-        # Once slewing or indexing is complete, state will go to IDLE
-        return self.current_state == self.IDLE
+        return (
+            self.current_state != self.SCANNING
+            and abs(self.motor.getClosedLoopError) < self.ACCEPTABLE_ERROR_COUNTS
```

Comment:
It runs fine in the simulator - but our simulator doesn't do anything so it never makes the calls! Correct me if I'm doing something wrong!

Reply:
Sigh. Our testing infrastructure is, um, weak. Even the CI just finds relatively innocuous stuff but not an undefined variable. I've configured pylint as a pre-commit for myself, and it finds a bit more than pyflakes, but I've had to disable a lot of its checks because of magicbot magic and lack of libraries on the mac, so my configuration isn't right for the CI. And it doesn't find everything anyway. Even the python compiler doesn't find some of these things, because python is so dynamic.

Reply:
What about the TICKS_TO_SETTLE thing? (I really should have made it a separate conversation.)

Reply:
I wonder if we are better off checking the current velocity as well with `getSelectedSensorVelocity()`  - it means we don't have to keep a running count of ticks when on target. We just check that we are in position and that our speed is low enough to assure us that we aren't moving (much, if at all).

Reply:
I wondered the same thing, which is why I thought I'd see what CTRE recommends, and they said to check for settling. At 10 ticks, our settling takes 200ms. They use a window to get velocity, which I believe is 100ms, but it might be settable. We could use 5 ticks to make ours 100 too, though their velocity is much more accurate as they sample every 1ms during the window, if I'm not mistaken, though accuracy alone won't make much difference. Now if they are constantly measuring velocity in a sliding window and just return the latest value immediately, we can avoid waiting any longer than necessary. I need to go now, but it might be worth looking up the details of the velocity call to see if it looks like a clear win.

Reply:
The docs confirm that velocity is constantly being calculated in a sliding window. The take the difference between now and 100ms ago and insert it into a 64 sample sliding window average. They do this every 1ms. So it's the way to go. I have pushed a commit using this velocity. We need to tune the threshold still.